File: src/main/java/com/hy/frame/view/MyGridView.java
package com.hy.frame.view;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.GridView;

/**
 * MyGridView ScrollView 冲突
 *
 * @author HeYan
 * @time 2017/5/24 14:01
 */
@Deprecated
public class MyGridView extends GridView {

    public MyGridView(Context context) {
        super(context);
    }

    public MyGridView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public MyGridView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int expandSpec = MeasureSpec.makeMeasureSpec(Integer.MAX_VALUE >> 2, MeasureSpec.AT_MOST);
        super.onMeasure(widthMeasureSpec, expandSpec);
    }
}

File: src/main/java/com/hy/frame/widget/MyGridView.kt
package com.hy.frame.widget

import android.content.Context
import android.util.AttributeSet
import android.widget.GridView

/**
 * MyGridView ScrollView 冲突
 * @author HeYan
 * @time 2017/5/24 14:01
 */
@Deprecated("不建议使用GridView")
class MyGridView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : GridView(context, attrs, defStyleAttr) {

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val expandSpec = MeasureSpec.makeMeasureSpec(Integer.MAX_VALUE shr 2, MeasureSpec.AT_MOST)
        super.onMeasure(widthMeasureSpec, expandSpec)
    }
}